lib/command/run.rb

```diff
@@ -308,6 +261,8 @@ def base_workload_specs(workload)
     end
 
     def build_job_start_yaml # rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+      _, container_spec = base_workload_specs(original_workload)
```

Comment:
Even though `container_spec` is only used in line 298, we need to call `base_workload_specs` here because it sets `@container`.

Reply:
sounds very hacky, and not sure that container_spec is needed.

Reply:
We do need it. See if 1ce60c4db193410852977c67ce4f76aaa074ee74 makes it better. I moved the `@container` assignment out of `base_workload_specs`.